standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1327,7 +1327,7 @@ public void visit(LeafNode node) throws MetaException {
         return;
       }
 
-      String colTypeStr = partitionKeys.get(partColIndex).getType();
+      String colTypeStr = MetaStoreUtils.getBaseNameOfColType(partitionKeys.get(partColIndex).getType());
```

Comment:
i think you can use `TypeInfoUtils.getTypeInfoFromTypeString(colTypeStr )`

Reply:
Similar implementation is present in TypeInfoUtils#getBaseName(String) which does this conversion. But this would require adding serde module into standalone-metastore which seems to me like an overkill for this fix.